### PR TITLE
Make `shell/platform/{darwin|embedder}/...` compatible with `.clang-tidy`.

### PR DESCRIPTION
**Pull request by matanlurey**

### CODE REVIEW

File: shell/platform/embedder/embedder.h [hunk 1]
```diff
@@ -2263,6 +2263,8 @@ typedef struct {
 
 #ifndef FLUTTER_ENGINE_NO_PROTOTYPES
 
+// NOLINTBEGIN(google-objc-function-naming)
```
Comment (cbracken) [inline shell/platform/embedder/embedder.h:2266]: Is there a way for us to not apply google-objc-function-naming to non-Objective-C headers?

We should apply it to headers under //flutter/shell/platform/darwin, but most of the repo should be treated as C/C++.

**Review by cbracken — COMMENTED** [review]
The changes under //flutter/shell/platform/darwin lgtm. I'd prefer we adjust the linting config for embedder.h so we're not running unwanted linters on it.

File: shell/platform/embedder/embedder.h (hunk 1 above)
Comment (matanlurey) [reply shell/platform/embedder/embedder.h:2266]: My guess is because this is imported by Obj-C code, it assumes it's Obj-C.

**Comment by matanlurey** [comment]
> The changes under //flutter/shell/platform/darwin lgtm. I'd prefer we adjust the linting config for embedder.h so we're not running unwanted linters on it.

I unfortunately don't think there is a configuration option for what we want.

I'm of the mind that the NOLINT markers are the lesser of evils here, though defer to you.

**Review by cbracken — APPROVED** [review]
lgtm; would be nice to find a way to tell the linter somehow that embedder.h is a C header.

**Pull request merged**